File: src/Web/WebAngular/src/app/domain/view-models/pipeline.view-model.ts
import { PipelineStatus } from "../enums/pipeline-status.enum";

export interface PipelineViewModel {
	id: string;
	name: string;
	description: string | null;
	status: PipelineStatus;
	createdBy: string;
	creationDate: string;
	updatedBy: string;
	lastUpdate: string;
}

File: src/Core/Houston.Application/ViewModel/PipelineViewModels/PipelineViewModel.cs
namespace Houston.Application.ViewModel.PipelineViewModels {
	public class PipelineViewModel {
		public Guid Id { get; set; }

		public string Name { get; set; } = null!;

		public string? Description { get; set; }

		public PipelineStatus Status { get; set; }

		public string CreatedBy { get; set; } = null!;

		public DateTime CreationDate { get; set; }

		public string UpdatedBy { get; set; } = null!;

		public DateTime LastUpdate { get; set; }
	}
}
